### PR TITLE
Information Retrieval (6/5)

### PR DESCRIPTION
**Pull request by lucadiliello**
# Before submitting

- [X] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Did you make sure to update the docs?   
- [X] Did you write any new necessary tests?  

## What does this PR do?
implements Fall Our rate for IR. Original discussione [here](https://github.com/PyTorchLightning/pytorch-lightning/pull/4991).

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/references/functional.rst [hunk 1]
```diff
@@ -266,3 +266,10 @@ retrieval_recall [func]
 
 .. autofunction:: torchmetrics.functional.retrieval_recall
     :noindex:
+
+
+retrieval_fall_out [func]
+~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Borda) [inline docs/source/references/functional.rst:272]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: docs/source/references/modules.rst [hunk 2]
```diff
@@ -350,6 +350,13 @@ RetrievalRecall
     :noindex:
 
 
+RetrievalFallOut
+~~~~~~~~~~~~~~~~~~
```
Comment (Borda) [inline docs/source/references/modules.rst:354]:
```suggestion
~~~~~~~~~~~~~~~~
```

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@lucadiliello mind adding an entry to changelog... 🐰

File: tests/retrieval/test_fallout.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+def _fallout_at_k(target: np.ndarray, preds: np.ndarray, k: int = None):
+    """
+    Didn't find a reliable implementation of Fall-out in Information Retrieval, so,
+    reimplementing here. See wikipedia for more information about definition.
+    """
+    assert target.shape == preds.shape
+    assert len(target.shape) == 1  # works only with single dimension inputs
+
+    if k is None:
+        k = len(preds)
```
Comment (SkafteNicki) [inline tests/retrieval/test_fallout.py:45]:
```suggestion
    k = len(preds) if k is None else k
```

File: tests/retrieval/test_fallout.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+from torchmetrics.functional.retrieval.fall_out import retrieval_fall_out
+from torchmetrics.retrieval.retrieval_fallout import RetrievalFallOut
+
+seed_all(42)
+
+
+def _fallout_at_k(target: np.ndarray, preds: np.ndarray, k: int = None):
+    """
+    Didn't find a reliable implementation of Fall-out in Information Retrieval, so,
+    reimplementing here. See wikipedia for more information about definition.
```
Comment (SkafteNicki) [inline tests/retrieval/test_fallout.py:39]: could you please link the wiki page in the docstring

File: torchmetrics/functional/retrieval/fall_out.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        >>> from  torchmetrics.functional import retrieval_fall_out
+        >>> preds = tensor([0.2, 0.3, 0.5])
+        >>> target = tensor([True, False, True])
+        >>> retrieval_fall_out(preds, target, k=2)
+        tensor(1.)
+    """
+    preds, target = _check_retrieval_functional_inputs(preds, target)
+
+    if k is None:
+        k = preds.shape[-1]
```
Comment (SkafteNicki) [inline torchmetrics/functional/retrieval/fall_out.py:48]:
```suggestion
    k = preds.shape[-1] if k is None else k
```

File: tests/retrieval/test_fallout.py (hunk 4 above)
Comment (lucadiliello) [reply tests/retrieval/test_fallout.py:39]: Yesss

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**